### PR TITLE
fix: load_image should support PIL Image

### PR DESCRIPTION
**Pull request by Laisky**
# What does this PR do?

`load_image` should support `PIL.Image`.

Fixes https://github.com/huggingface/diffusers/pull/6479#issuecomment-1933440898

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
  - https://github.com/huggingface/diffusers/pull/6479
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/diffusers/utils/loading_utils.py [hunk 1]
```diff
@@ -32,6 +32,8 @@ def load_image(
             raise ValueError(
                 f"Incorrect path or URL. URLs must start with `http://` or `https://`, and {image} is not a valid path."
             )
+    elif isinstance(image, PIL.Image.Image):
+        image = image
```
Comment (sayakpaul) [inline src/diffusers/utils/loading_utils.py:36]: Wonder why would someone call `load_image()` on an already properly loaded image, though.

Comment (Laisky) [reply src/diffusers/utils/loading_utils.py:36]: Many tutorials are written like this😁, Perhaps to ensure things like RGB

**Comment by sayakpaul** [comment]
Hmm as per the method signature, it seems like `load_image()` should support a PIL image but the code below currently doesn't work:

```python
from diffusers.utils import load_image 

image = load_image(image)
```

where `image` is a PIL image. I'd be in favor of this. @yiyixuxu WDYT?

**Comment by yiyixuxu** [comment]
@sayakpaul 
yes we should add it back
I think this is in the original implementation anyway

**Comment by sayakpaul** [comment]
Alright. Feel free to merge once the CI is green.

**Pull request merged**